open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/manage/impl/VRPConsentRequestHandler.java

```diff
@@ -0,0 +1,210 @@
+                                consent.getClientID(), consentManageData.getClientId()));
+                    }
+                    throw new ConsentException(ResponseStatus.BAD_REQUEST,
+                            ErrorConstants.NO_CONSENT_FOR_CLIENT_ERROR);
+                }
+
+                JSONObject receiptJSON = (JSONObject) new JSONParser(JSONParser.MODE_PERMISSIVE).
+                        parse(consent.getReceipt());
+                consentManageData.setResponsePayload(ConsentManageUtil
+                        .getInitiationRetrievalResponse(receiptJSON, consent, consentManageData,
+                                ConsentExtensionConstants.VRP));
+                consentManageData.setResponseStatus(ResponseStatus.OK);
+            } catch (ConsentManagementException | ParseException e) {
+                log.error(ErrorConstants.INVALID_CLIENT_ID_MATCH);
+                throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR,
+                        ErrorConstants.ACC_INITIATION_RETRIEVAL_ERROR);
+            }
+        } else {
+            log.error(ErrorConstants.INVALID_CONSENT_ID);
+            throw new ConsentException(ResponseStatus.BAD_REQUEST, ErrorConstants.INVALID_CONSENT_ID);
+        }
+    }
+
+    /**
+     * Handles the DELETE request for revoking or deleting a consent.
+     * @param consentManageData   Object containing request details
+     */
+    @Override
+    public void handleConsentManageDelete(ConsentManageData consentManageData) {
+
+        ConsentManageUtil.handleConsentManageDelete(consentManageData);
+    }
+
+    /**
+     * Method to handle Variable Recurring Payment POST requests.
+     * This private method processes Variable Recurring Payment POST requests, creating a new consent
+     * based on the provided request payload. It performs the following actions:
+     * - Creates a DetailedConsentResource representing the consent initiation.
+     * - Stores consent attributes, including the idempotency key.
+     * - Constructs the response payload containing initiation details and sets appropriate headers.
+     * - Sets the response status to Created.
+     *
+     * @param consentManageData Object containing request details, including client ID, request payload, headers.
+     */
+    public void handlePaymentPost(ConsentManageData consentManageData, Object request)
+            throws ConsentManagementException {
+
+        JSONObject requestObject = (JSONObject) request;
+
+        // Create a ConsentResource representing the requested consent
+        ConsentResource requestedConsent = new ConsentResource(consentManageData.getClientId(),
+                requestObject.toJSONString(), ConsentExtensionConstants.VRP,
+                ConsentExtensionConstants.AWAITING_AUTH_STATUS);
+
+        // Create the consent
+        DetailedConsentResource createdConsent = ConsentExtensionsDataHolder.getInstance().getConsentCoreService()
```

Comment:
We can replace `ConsentExtensionsDataHolder.getInstance().getConsentCoreService()` with `ConsentServiceUtil.getConsentService()`

Reply:
Fixed in [3205f1b]